static/js/lib/learning_resources.js

```diff
@@ -27,24 +32,130 @@ export const availabilityLabel = (availability: ?string) => {
   }
 }
 
+export const availabilityFilterToMoment = (filter: string, ending: boolean) => {
+  // Convert an Elasticsearch date_range filter string to a moment
+  const format = /(now)(\+)?(\d+)?([Md])?/
+  const match = format.exec(filter)
+  if (match) {
+    let dt = moment()
+    if (match[3] && match[4]) {
+      dt = dt.add(match[3], match[4] === "d" ? "days" : "months")
+    }
+    if (ending) {
+      // $FlowFixMe: this is fine according to moment docs, and it works
+      dt.set({ hour: 23, minute: 59, second: 59 })
+    } else {
+      // $FlowFixMe: this is fine according to moment docs, and it works
+      dt.set({ hour: 0, minute: 0, second: 0 })
+    }
+    return dt
+  }
+}
+
+export const inDateRanges = (run: CourseRun, availabilities: Array<string>) => {
+  let inRange = false
+  availabilities.forEach(availability => {
+    if (AVAILABILITY_MAPPING[availability]) {
+      const from = availabilityFilterToMoment(
+        AVAILABILITY_MAPPING[availability].filter.from,
+        false
+      )
+      const to = availabilityFilterToMoment(
+        AVAILABILITY_MAPPING[availability].filter.to,
+        true
+      )
+      const startDate = runStartDate(run)
+      if (
+        (isNil(from) ||
+          // $FlowFixMe: if we get this far, only moments are compared
+          startDate.isSameOrAfter(from)) &&
+        (isNil(to) ||
+          // $FlowFixMe: if we get this far, only moments are compared
```

Comment:
I think you can remove these flow disables here if you switch `isNil(from)` to `!from`

Reply:
✔️